src/react/components/InterfaceEditor.tsx

```diff
@@ -0,0 +1,945 @@
+  use_ttl: 'Use TTL',
+};
+
+const MappingRow = ({ className, mapping, onEdit, onDelete }: MappingRowProps) => {
+  return (
+    <Accordion>
+      <Card className={className}>
+        <Accordion.Toggle
+          eventKey={mapping.endpoint}
+          as={Card.Header}
+          className="d-flex align-items-center flex-wrap"
+        >
+          <span className="flex-grow-1">
+            <Badge variant="secondary">{mapping.type}</Badge>
+            <Button className="text-left text-truncate" variant="link">
+              {mapping.endpoint}
+            </Button>
+          </span>
+          <Button className="mr-2" variant="outline-primary" onClick={onEdit}>
+            Edit...
+          </Button>
+          <Button variant="outline-primary" onClick={onDelete}>
+            Remove
+          </Button>
+        </Accordion.Toggle>
+        <Accordion.Collapse eventKey={mapping.endpoint}>
+          <Card.Body>
+            {mapping.description && (
+              <>
+                <h5>Description</h5>
+                <p>{mapping.description}</p>
+              </>
+            )}
+            {mapping.documentation && (
+              <>
+                <h5>Documentation</h5>
+                <p>{mapping.documentation}</p>
+              </>
+            )}
+            {mapping.allowUnset && (
+              <>
+                <h5>Allow Unset</h5>
+                <p>True</p>
+              </>
+            )}
+            {mapping.explicitTimestamp && (
+              <>
+                <h5>Explicit Timestamp</h5>
+                <p>True</p>
+              </>
+            )}
+            {mapping.reliability && (
+              <>
+                <h5>Reliability</h5>
+                <p>{reliabilityToLabel[mapping.reliability]}</p>
+              </>
+            )}
+            {mapping.retention && (
+              <>
+                <h5>Retention</h5>
+                <p>{retentionToLabel[mapping.retention]}</p>
+              </>
+            )}
+            {mapping.expiry && (
+              <>
+                <h5>Expiry</h5>
+                <p>{mapping.expiry} seconds</p>
+              </>
+            )}
+            {mapping.databaseRetentionPolicy && (
+              <>
+                <h5>Database Retention</h5>
+                <p>{databaseRetentionToLabel[mapping.databaseRetentionPolicy]}</p>
+              </>
+            )}
+            {mapping.databaseRetentionTtl != null && (
+              <>
+                <h5>Database Retention TTL</h5>
+                <p>{mapping.databaseRetentionTtl} seconds</p>
+              </>
+            )}
+          </Card.Body>
+        </Accordion.Collapse>
+      </Card>
+    </Accordion>
+  );
+};
+
+const getDefaultMapping = (params: {
+  interfaceType: AstarteInterface['type'];
+  interfaceAggregation: AstarteInterface['aggregation'];
+}): AstarteMapping => {
+  if (params.interfaceType === 'datastream' && params.interfaceAggregation === 'individual') {
```

Comment:
what about
```
    return {
      endpoint: '',
      type: 'double',
      explicitTimestamp: params.interfaceType === 'datastream'
    };
```
if false, it won't be shown anyway

Reply:
It's a conceptual difference. If I understood the documentation correctly, `explicitTimestamp` doesn't make sense for Properties interfaces. If that's the case, I prefer not to define it altogether.